airflow/providers/openlineage/utils/utils.py

```diff
@@ -78,11 +78,38 @@ def get_job_name(task: TaskInstance) -> str:
 
 
 def get_custom_facets(task_instance: TaskInstance | None = None) -> dict[str, Any]:
+    from airflow.providers.openlineage.extractors.manager import try_import_from_string
+
     custom_facets = {}
     # check for -1 comes from SmartSensor compatibility with dynamic task mapping
     # this comes from Airflow code
     if hasattr(task_instance, "map_index") and getattr(task_instance, "map_index") != -1:
         custom_facets["airflow_mappedTask"] = AirflowMappedTaskRunFacet.from_task_instance(task_instance)
+
+    # Append custom run facets by executing the custom_run_facets.
+    for custom_facet_func in conf.custom_run_facets():
+        func: type[function] | None = try_import_from_string(custom_facet_func)
+        if not func:
+            log.warning(
+                "OpenLineage is unable to import custom facet function `%s`; will ignore it.",
+                custom_facet_func,
+            )
+            continue
+        facet: dict[str, BaseFacet] = func(task_instance)
```

Comment:
I noticed that in examples here we are using attrs.asdict(), so we are providing the facets as dicts, but here we actually expect type[BaseFacet]. Could you adjust the typing and examples so that it matches?

Reply:
And one more question / decision to make: should we wrap the function call with try/except to avoid crashing the whole event emission or do we let the user crash it if they provided the function that breaks? I would vote for putting a try/except with some descriptive logging here.

Reply:
>And one more question / decision to make: should we wrap the function call with try/except to avoid crashing the whole event emission or do we let the user crash it if they provided the function that breaks? I would vote for putting a try/except with some descriptive logging here.

Makes sense. Added try-except in db25be02a92c2f0f81b27710d92fb760365f2148.

Reply:
>I noticed that in examples here we are using attrs.asdict(), so we are providing the facets as dicts, but here we actually expect type[BaseFacet]. Could you adjust the typing and examples so that it matches?

We expect that the functions return `dict` or `None`. Addressed the typing error in 3790b7410a667b2cfe691b8a775d3ac0e31c6fc8.

Reply:
I was more concerned about the type of the dict's key, so whether the function should produce:
`facet: dict[str, Type[BaseFacet]] | None` or `facet: dict[str, dict] | None` ? Maybe it doesn't matter, but then we should probably also let the users know 😉

Reply:
Return type can be `facet: dict[str, dict] | None`. Addressed in b5d3f37b5aa4171d112823ade8c41bf12743ef8b.